### PR TITLE
feat: Config validator

### PR DESCRIPTION
**Pull request by L0wry**
test coverage 
configValidator.js           |      100 |    91.67 |      100 |      100 |                46 |

info [configValidator] : ❗️Please add missing mandatory fields to your config:
info [configValidator] : browser
info [config Validator] :  Exiting Aye Spy
info [configValidator] : Config validated ✅

### CODE REVIEW

File: src/bin/run.js [hunk 1]
```diff
@@ -29,17 +40,22 @@ program
   .option('c, --config [config]', 'Path to your config')
   .option('r, --remote', 'Upload new baseline to remote storage')
   .action(async options => {
-    const configPath = path.resolve(options.config);
-    const config = require(path.resolve(options.config)); // eslint-disable-line import/no-dynamic-require
+    try {
+      const config = require(path.resolve(options.config)); // eslint-disable-line import/no-dynamic-require
+
+      if (!config) throw Error(`Could not resolve config at path: ${config}`);
```
Comment (Xin00163) [inline src/bin/run.js]: ???

File: src/configValidator.test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  isLocalConfigValid,
+  isRemoteConfigValid
+} from './configValidator';
+
+describe('The Config Validator', () => {
+  it('local config returns false for invalid configs', () => {
+    const config = {
+      gridUrl: 'http://selenium.com:4444/wd/hub'
+    };
+    expect(isLocalConfigValid(config)).toBe(false);
```
Comment (Xin00163) [inline src/configValidator.test.js]: we should mock logger and assert that the correct tings are reported

File: src/configValidator.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    missingConfigFields.push('env variable: AWS_SECRET_ACCESS_KEY');
+  if (!process.env.AWS_ACCESS_KEY_ID)
+    missingConfigFields.push('env variable: AWS_ACCESS_KEY_ID');
+
+  return isValid(missingConfigFields);
+}
+
+function isLocalConfigValid(config) {
+  const missingConfigFields = [
+    'gridUrl',
```
Comment (Xin00163) [inline src/configValidator.js]: declare mandatory fields seperatelhy fsigjsarhri

File: src/configValidator.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import logger from './logger';
+
+function isValid(missingConfigFields) {
+  if (missingConfigFields.length > 0) {
+    logger.info(
+      'configValidator',
+      `❗️Please add missing mandatory fields to your config: \n${missingConfigFields.toString()}`
```
Comment (mdcruz) [inline src/configValidator.js]: Add space between emoji and the word please. It looks slightly off on the reporting

File: src/configValidator.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    if (isRemote) {
+      isRemoteConfigCorrect = isRemoteConfigValid(config);
+    }
+
+    if (isLocalConfigValid(config) && isRemoteConfigCorrect) {
+      logger.info('configValidator', 'Config validated ✅');
+      resolve();
+    } else {
+      logger.info(
+        'config Validator',
```
Comment (mdcruz) [inline src/configValidator.js]: rename to configValidator to be consistent

File: src/bin/run.js (hunk 1 above)
Comment (kaliabadi) [reply src/bin/run.js]: yeah wtf Matt

File: src/configValidator.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+function isLocalConfigValid(config) {
+  const missingConfigFields = [
+    'gridUrl',
+    'baseline',
+    'latest',
+    'generatedDiffs',
+    'report',
+    'scenarios',
+    'browser'
```
[the remark this answers is not in the document]
Comment (L0wry) [reply src/configValidator.js]: browser is usually passed on the cli, but also is able to be passed through the config.
Probably should update the readme to make this obvious.

There are plans for a scenario validator, but I wanted to do it as a separate PR.

Height and width are no longer used. Use the viewports array - which also needs a validator.
Default sizes are also set when the snapshotter is instantiated

File: src/configValidator.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  beforeEach(() => {
+    logger.info = jest.fn();
+  });
+
+  it('local config returns false for invalid configs', () => {
+    const config = {
+      gridUrl: 'http://selenium.com:4444/wd/hub'
+    };
+
+    logger.info = jest.fn();
```
Comment (mdcruz) [inline src/configValidator.test.js]: remove duplicate code as this is already mock on beforeEach

**Pull request merged**